compiler/lib/driver.mli

```diff
@@ -20,16 +20,22 @@
 
 type profile
 
+type 'a target =
```

Comment:
I'm not convinced by the use of GADT here.
Can't you just export a new function with what you need ?

Reply:
Indeed, I think one can define a function `optimize` that corresponds to the initial part of function `full` and use it to define a Wasm-specific funtion.

Reply:
Done